<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Linear System Solver</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icons8_Abacus_50px.png</normaloff>icons8_Abacus_50px.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLineEdit" name="insert_equation_text">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>40</y>
      <width>270</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#insert_equation_text{border: 1px solid rgb(81, 80, 80);border-radius: 10px; padding: 0 8px; selection-background-color: darkgray;font-size: 13px;}#insert_equation_text:focus { color:black;background-color:rgb(59, 255, 38);}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QComboBox" name="selected_method">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>220</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Gaussian-elimination</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>LU decomposition</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Gaussian-Jordan</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Gauss-Seidel</string>
     </property>
    </item>
   </widget>
   <widget class="QTableWidget" name="equations_table_widget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>70</y>
      <width>270</width>
      <height>261</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::SingleSelection</enum>
    </property>
    <property name="rowCount">
     <number>1</number>
    </property>
    <property name="columnCount">
     <number>1</number>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>true</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>270</number>
    </attribute>
    <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
     <bool>false</bool>
    </attribute>
    <row/>
    <column>
     <property name="text">
      <string>Equations</string>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <italic>true</italic>
       <bold>true</bold>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignTop</set>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>340</y>
      <width>100</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>Epislon:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>340</y>
      <width>100</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>Max Iterations:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="epislon_criteria_text">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>360</y>
      <width>130</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#epislon_criteria_text{border: 1px solid rgb(81, 80, 80);border-radius: 10px; padding: 0 8px; selection-background-color: darkgray;font-size: 13px;}
#epislon_criteria_text:focus { color:white;
background-color:rgb(59, 255, 38);}</string>
    </property>
    <property name="text">
     <string>0.00001</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="max_iterations_criteria_text">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>360</y>
      <width>130</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#max_iterations_criteria_text{border: 1px solid rgb(81, 80, 80);border-radius: 10px; padding: 0 8px; selection-background-color: darkgray;font-size: 13px;}
#max_iterations_criteria_text:focus { color:white;
background-color:rgb(59, 255, 38);}</string>
    </property>
    <property name="text">
     <string>50</string>
    </property>
   </widget>
   <widget class="QPushButton" name="insert_equation_button">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>40</y>
      <width>50</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#insert_equation_button{border: 1px solid black;border-radius:10px;}#insert_equation_button:hover{border: 1px solid aqua;}#insert_equation_button:pressed{background-color: aqua;}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>icons8_Checkmark_52px_3.png</normaloff>icons8_Checkmark_52px_3.png</iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="delete_equation_button">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>70</y>
      <width>50</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#delete_equation_button{border: 1px solid black;border-radius:10px;}#delete_equation_button:hover{border: 1px solid aqua;}#delete_equation_button:pressed{background-color: aqua;}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>icons8_Delete_48px.png</normaloff>icons8_Delete_48px.png</iconset>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>400</y>
      <width>100</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>Notification:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="notification_text">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>420</y>
      <width>270</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="adjust_criteria_button">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>360</y>
      <width>50</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#adjust_criteria_button{border: 1px solid black;border-radius:10px;}#adjust_criteria_button:hover{border: 1px solid aqua;}#adjust_criteria_button:pressed{background-color: aqua;}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>icons8_Checkmark_52px_3.png</normaloff>icons8_Checkmark_52px_3.png</iconset>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>50</width>
      <height>500</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(81, 80, 80)</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="output_file_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>10</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#output_file_button{border: none}#output_file_button:hover{border: 1px solid aqua;border-radius:10px;}#output_file_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Check_File_50px.png</normaloff>icons8_Check_File_50px.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="input_file_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>50</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#input_file_button{border: none}#input_file_button:hover{border: 1px solid aqua;border-radius:10px;}#input_file_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Add_File_64px.png</normaloff>icons8_Add_File_64px.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="clear_equations_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>420</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#clear_equations_button{border: none}#clear_equations_button:hover{border: 1px solid aqua;border-radius:10px;}#clear_equations_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Trash_64px.png</normaloff>icons8_Trash_64px.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="plot_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>90</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#plot_button{border: none}#plot_button:hover{border: 1px solid aqua;border-radius:10px;}#plot_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Plot_64px.png</normaloff>icons8_Plot_64px.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="steps_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>130</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#steps_button{border: none}#steps_button:hover{border: 1px solid aqua;border-radius:10px;}#steps_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Analyze_50px.png</normaloff>icons8_Analyze_50px.png</iconset>
     </property>
    </widget>
    <widget class="QPushButton" name="clear_all_button">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>460</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">#clear_all_button{border: none}#steps_button:hover{border: 1px solid aqua;border-radius:10px;}#clear_all_button:pressed{background-color: aqua;}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <normaloff>icons8_Broom_64px.png</normaloff>icons8_Broom_64px.png</iconset>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>441</x>
      <y>10</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Linear system solution:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="solve_button">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>460</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#solve_button{border: 1px solid white;border-radius:10px;color:white;
background-color:rgb(255, 48, 51);}#solve_button:hover{border: 1px solid rgb(255, 48, 51);color:rgb(255, 48, 51);background-color:white;}#solve_button:pressed{border: 1px solid white;border-radius:10px;color:white;
background-color:rgb(255, 48, 51);}</string>
    </property>
    <property name="text">
     <string>Solve</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="solution_table_widget">
    <property name="geometry">
     <rect>
      <x>431</x>
      <y>40</y>
      <width>270</width>
      <height>260</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <property name="rowCount">
     <number>1</number>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>135</number>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>40</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>20</number>
    </attribute>
    <attribute name="verticalHeaderMinimumSectionSize">
     <number>25</number>
    </attribute>
    <row/>
    <column>
     <property name="text">
      <string>Variable</string>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <italic>true</italic>
       <bold>true</bold>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Solution</string>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <italic>false</italic>
       <bold>true</bold>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>150</y>
      <width>750</width>
      <height>80</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:white;</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>0</y>
      <width>80</width>
      <height>500</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:white;</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>431</x>
      <y>320</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Execution time:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="execution_time_text">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>320</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QTableWidget" name="insert_initial_guesses_table_widget">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>360</y>
      <width>270</width>
      <height>91</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <property name="verticalScrollMode">
     <enum>QAbstractItemView::ScrollPerItem</enum>
    </property>
    <property name="showGrid">
     <bool>true</bool>
    </property>
    <property name="gridStyle">
     <enum>Qt::SolidLine</enum>
    </property>
    <property name="cornerButtonEnabled">
     <bool>true</bool>
    </property>
    <property name="columnCount">
     <number>1</number>
    </property>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>30</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>25</number>
    </attribute>
    <attribute name="verticalHeaderMinimumSectionSize">
     <number>20</number>
    </attribute>
    <row>
     <property name="text">
      <string>Variable</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Initial guesses</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </row>
    <column/>
   </widget>
   <widget class="QLineEdit" name="insert_initial_guesses_text">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>460</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#insert_intial_guesses_text{border: 1px solid rgb(81, 80, 80);border-radius: 10px; padding: 0 8px; selection-background-color: darkgray;font-size: 13px;}
#insert_intial_guesses_text:focus { color:white;
background-color:rgb(59, 255, 38);}</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>Guesses</string>
    </property>
   </widget>
   <widget class="QPushButton" name="insert_initial_guesses_button">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>460</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#insert_initial_guesses_button{border: 1px solid black;border-radius:10px;}#insert_initial_guesses_button:hover{border: 1px solid aqua;}#insert_initial_guesses_button:pressed{background-color: aqua;}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>icons8_Checkmark_52px_3.png</normaloff>icons8_Checkmark_52px_3.png</iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="delete_initial_guess_button">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>460</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#delete_initial_guess_button{border: 1px solid black;border-radius:10px;}#delete_initial_guess_button:hover{border: 1px solid aqua;}#delete_initial_guess_button:pressed{background-color: aqua;}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>icons8_Delete_48px.png</normaloff>icons8_Delete_48px.png</iconset>
    </property>
   </widget>
   <zorder>frame_3</zorder>
   <zorder>frame_2</zorder>
   <zorder>frame</zorder>
   <zorder>insert_equation_text</zorder>
   <zorder>selected_method</zorder>
   <zorder>equations_table_widget</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>epislon_criteria_text</zorder>
   <zorder>max_iterations_criteria_text</zorder>
   <zorder>insert_equation_button</zorder>
   <zorder>delete_equation_button</zorder>
   <zorder>label_3</zorder>
   <zorder>notification_text</zorder>
   <zorder>adjust_criteria_button</zorder>
   <zorder>label_4</zorder>
   <zorder>solve_button</zorder>
   <zorder>solution_table_widget</zorder>
   <zorder>label_6</zorder>
   <zorder>execution_time_text</zorder>
   <zorder>insert_initial_guesses_table_widget</zorder>
   <zorder>insert_initial_guesses_text</zorder>
   <zorder>insert_initial_guesses_button</zorder>
   <zorder>delete_initial_guess_button</zorder>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
